magefiles/rulesengine/readme.md

```diff
@@ -0,0 +1,211 @@
+				return true
+			}
+			return false
+		}),
+		Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+			rctx.LabelFilter = "e2e-demo,rhtap-demo,spi-suite,remote-secret,integration-service,ec,build-templates,multi-platform"
+			return ExecuteTestAction(rctx)
+		})}},
+
+} 
+ 
+ ```
+
+ ### Registering a rule catalog to the engine
+
+ Here I've created a catagory called tests within the MageEngine and I've assigned infra-deployments catalog to the infra-deployments key
+ under the tests category
+
+ ```go
+ var MageEngine = rulesengine.RuleEngine{
+	"tests": {
+		"infra-deployments": testselection.InfraDeploymentsTestRulesCatalog,
+	},
+}
+ ```
+
+ ### Executing the MageEngine within a mage file
+
+ This will cause the engine to load and iterate over the rules of the category test and apply any that had their conditions match.
+
+ ```go
+
+    rctx := rulesengine.NewRuleCtx()
+	rctx.DryRun = true
+
+	err = engine.MageEngine.RunRulesOfCategory("tests", rctx)
+ ```
+
+ ### Creating a complex rule using a mixture of conditional filters
+
+ In this example, the release-catalog-repo has a slightly more complex set of rules where depending on certain conditions
+ met by the CI job the test should execute with a different set of ginkgo label filters. So you will see a mixture of the 
+ Any/All/None conditional filters
+
+ The first rule's logic is: WHEN EITHER RepoName equals release-service-catalog AND NOT pr paired OR RepoName equals release-service-catalog
+ AND is a rehearse job AND NOT pr paired THEN run ginkgo with release-pipelines label filter
+
+ The second rule's logic is: WHEN RepoName equals release-service-catalog AND pr is paired AND NOT a rehearse job THEN run ginkgo with 
+ release-pipelines && !fbc-tests label filters
+
+
+ ```go
+
+ rulesengine.Rule{Name: "Release Catalog Test Execution",
+		Description: "Runs release catalog tests on release-service-catalog repo on pull/rehearsal jobs.",
+		Condtion:    rulesengine.Any{rulesengine.All{rulesengine.ConditionFunc(releaseCatalogRepoCondition), rulesengine.None{rulesengine.ConditionFunc(isPaired)}}, rulesengine.All{rulesengine.ConditionFunc(releaseCatalogRepoCondition), rulesengine.ConditionFunc(isRehearse)}, rulesengine.None{rulesengine.ConditionFunc(isPaired)}},
+		Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+			rctx.LabelFilter = "release-pipelines"
+			return ExecuteTestAction(rctx)
+		})}},
+rulesengine.Rule{Name: "Release Catalog PR paired Test Execution",
+		Description: "Runs release catalog tests except for the fbc tests on release-service-catalog repo when PR paired and not a rehearsal job.",
+		Condtion:    rulesengine.All{rulesengine.ConditionFunc(releaseCatalogRepoCondition), rulesengine.ConditionFunc(isPaired), rulesengine.None{rulesengine.ConditionFunc(isRehearse)}},
+		Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
```

Comment:
Need to add this [part](https://github.com/konflux-ci/e2e-tests/blob/main/magefiles/magefile.go#L503C1-L510C66) in the Actions later. It will be registered as action, right?

Reply:
In theory, Yes, that is how it would work! But talking with Pavel setting those variables in this Rule is too late since bootstrapping depends on these variables.